### PR TITLE
Design backend

### PR DESCRIPTION
**Pull request by yoannLafore**
This PR adds the parts starting from **Backend** in the *Design and Implementation* section.

### CODE REVIEW

File: template/9-design.md [hunk 1]
```diff
@@ -22,40 +22,112 @@
...
+
+When a listenable logic is required, the cloud function will create a temporary document in a specific collection that the client will listen to. To maintain consistency with our *Database Interaction* policy, the client will only have read access to these documents, and any writes will be performed through backend API calls. This ensures that the client is promptly notified of relevant events while preserving the integrity and security of the backend.
+
+
+### Framework
+
+The backend for Proxima will utilize the Firebase suite. Specifically, the database will be managed through Firestore and the media storage will be held on Firebase Storage solution. The backend API will be implemented using Firebase Cloud Functions, written in Node.js as required by the framework. These functions will be callable from the client. This setup provides a scalable and efficient infrastructure for managing the application's backend operations, ensuring reliability and performance.
```
Comment (TheTexanCodeur) [inline template/9-design.md:48]:
```suggestion
The backend for Proxima will utilize the Firebase suite. Specifically, the database will be managed through Firestore and the media storage will be held on the Firebase Storage solution. The backend API will be implemented using Firebase Cloud Functions, written in Node.js as required by the framework. These functions will be callable from the client. This setup provides a scalable and efficient infrastructure for managing the application's backend operations, ensuring reliability and performance.
```

Comment (TheTexanCodeur) [inline template/9-design.md:48]:
```suggestion
The backend for Proxima will utilize the Firebase suite. Specifically, the database will be managed through Firestore and the media storage will be held on Firebase Storage Solution. The backend API will be implemented using Firebase Cloud Functions, written in Node.js as required by the framework. These functions will be callable from the client. This setup provides a scalable and efficient infrastructure for managing the application's backend operations, ensuring reliability and performance.
```

File: template/9-design.md [hunk 2]
```diff
@@ -22,40 +22,112 @@
...
+
 
 ## Data Model
 
-*What data are you collecting / managing?*
+The data for Proxima are stored on Firebase Firestore in a NoSQL, document-oriented database. Additionally, data are cached locally on the client device to allow for offline usage. The data are organized into collections representing the components necessary for the application. Due to the nature of Proxima, the data are tightly tied to the user; therefore, authentication is required to access any of the data. This ensures that user data is secure and accessible only to authorized users.
```
Comment (TheTexanCodeur) [inline template/9-design.md:53]:
```suggestion
The data for Proxima is stored on Firebase Firestore in a NoSQL, document-oriented database. Additionally, data are cached locally on the client device to allow for offline usage. The data are organized into collections representing the components necessary for the application. Due to the nature of Proxima, the data are tightly tied to the user; therefore, authentication is required to access any of the data. This ensures that user data is secure and accessible only to authorized users.
```

File: template/9-design.md (hunk 1 above)
Comment (TheTexanCodeur) [reply template/9-design.md:48]: I'm not sure of the phrasing here. Are you talking about "Firebase Storage Solution" ? If not, could we  maybe rephrase the end of this sentence like this : "... will be held on the Firebase Storage solution"? What do you think?

Comment (yoannLafore) [reply template/9-design.md:48]: That's a very good point indeed !

After looking it up, I believe that the actual name is Firebase Cloud Storage.
Thus, I propose that I replace `Firebase Storage solution` by `Firebase Cloud Storage`. What do you think ?

Comment (TheTexanCodeur) [reply template/9-design.md:48]: Yes perfect 👍🏼

Comment (yoannLafore) [reply template/9-design.md]: Nice thanks !

Done in d6a3993ed41a90094e1348304bbf9ee9be19edc0 :+1:

**Review by TheTexanCodeur — APPROVED** [review]
LGTM 🚀

**Comment by yoannLafore** [comment]
Thank you for your great suggestions :tada:
